### DISCUSSION TITLE
using prism for code formatting

### DISCUSSION THREAD

**Discussion by arunkumar413** [PrismJS/prism#3516] [Q&A]
Hello Team,

I'd like to understand if prism can be used for formatting  code like prettier. Seems like prism returns the HTML markup as well to display the code. Is there a way to make prism return only the formatted code? 

Thanks,
Arun

**Comment by RunDevelopment** [comment]
Hello!

I'm not sure what you mean by "formatting". Prism is a syntax highlighter, so we don't change source text, we only add markup around it to give it different styles. We do offer the [normalize whitespace plugin](https://prismjs.com/plugins/normalize-whitespace/), but that can hardly be called a formatter. We definitely can't do formatting like Prettier does.

**Reply by arunkumar413** [reply]
By formatting I mean similar to what prettier vs code extension does to the code.

**Reply by RunDevelopment** [reply]
No, Prism can't do that.
